ui-admin/src/user/EditPortalUserModal.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react'
+import { Modal } from 'react-bootstrap'
+import Api from 'api/api'
+import { Role } from 'api/adminUser'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { successNotification } from 'util/notifications'
+import { Store } from 'react-notifications-component'
+import { doApiLoad, useLoadingEffect } from '../api/api-utils'
+import { RoleSelector } from './AdminUserDetail'
+
+/** creates a new admin user */
+const EditPortalUserModal = ({ userId, portalShortcode, onDismiss, userUpdated }:
+  {
+    userId: string,
+    portalShortcode: string,
+    onDismiss: () => void,
+    userUpdated: () => void
+  }) => {
+  const [isLoading, setIsLoading] = useState(false)
+
+  /** there might be a case for supporting updating usernames, but for now we only support updating roles */
+  const [updateParams, setUpdateParams] = useState<{username: string, roleNames: string[]}>({
+    username: '',
+    roleNames: []
+  })
+  const [roles, setRoles] = useState<Role[]>([])
+
+  useLoadingEffect(async () => {
+    const [allRoles, loadedUser] = await Promise.all([
+      Api.fetchRoles(),
+      Api.fetchAdminUser(userId, portalShortcode)
+    ])
+    setUpdateParams({
+      username: loadedUser.username,
+      roleNames: loadedUser.portalAdminUsers![0].roles.map(role => role.name)
+    })
+    setRoles(allRoles)
+  })
+
+  const updateUser = async () => {
+    doApiLoad(async () => {
+      await Api.updatePortalUser(portalShortcode, userId, updateParams.roleNames)
```

Comment:
Since this patch payload includes the entire list of roles that should be persisted, adding/removing one role from another user requires the admin to have _all_ of the final roles.

For example, I can't remove `prototype_access` from a user if they also have `study_publisher` and I'm not also a `study_publisher` myself

We probably want the backend to sort out the diff and only reject the call if they're actually modifying a role that they don't have access to.